src/doc/en/developer/coding_in_python.rst

```diff
@@ -36,11 +37,24 @@ using one of two mechanisms:
 - Backport packages
 
   - `importlib_metadata <../reference/spkg/importlib_metadata>`_
-    (to be used in place of ``importlib.metadata``),
-  - `importlib_resources <../reference/spkg/importlib_resources>`_
-    (to be used in place of ``importlib.resources``),
+    (to be used in place of ``importlib.metadata`` when Python older
+    than 3.11 is in use),
+  - `importlib_resources <../reference/spkg/importlib_resources>`_:
+    If necessary, you can use it in place of ``importlib.resources``
+    when Python older than 3.11 is in use if you need a newer
+    feature::
+
+        import sys
+
+        if sys.version_info < (3, 11):
+            # Use backport package providing Python 3.11 features
+            from importlib_resources import files
+        else:
+            from importlib.resources import files
+
   - `typing_extensions <../reference/spkg/typing_extensions>`_
-    (to be used in place of ``typing``).
+    (to be used in place of ``typing`` when any of the features
+    introduced after Python 3.9.0 are used).
```

Comment:
You forgot about this, other that this the PR looks good to me.

I would remove almost all of this section, starting from "Some key language and library functions..." to the end, but I understand if you don't want to do it here.

Reply:
I didn't forget about it. I actually changed the language so it does not unconditionally asks for it to be imported like that.

Reply:
Please, let's not play games.

Two reviewers showed concern for this. You can have a different opinion, but if you accept to take it out "to regain velocity", then do it without tricks. I already stated my positive review conditional on this.

Reply:
Gonzalo, this is uncalled for.

Reply:
What is your concern about this improvement of the documentation in the "Language Standard" section?

Reply:
> What is your concern about this improvement

This is your opinion, reviewers don't agree, please respect our point of view.

As it was mentioned before, some distributions don't ship `importlib_resources` anymore for good reasons so we should not use it (we don't) nor instruct people to use it.

AFAICT there is no use of any of the backport packages in sagemath. The fact that importlib.resources doesn't support namespace packages is minor because:
 a. it will be supported in python 3.9 which is not very far away
 b. for the time being we can either place package data in normal packages, or just use the old method as in `os.path.join(os.path.dirname(__file__), 'sage-maxima.lisp')`, this one actually makes it for a good example because it's in a namespace package.

NOTE: the main difference of using `importlib.resources` instead of the old way is that the former supports using package data from a zip file (i.e. directly from a wheel). However, running from zip files doesn't support extensions since dynamic libraries can't be `dlopen`ed from a zip file, so sagemath will never be able to run directly from wheels.

Reply:
Gonzalo, even the toned down version ("if you accept to take it out 'to regain velocity', then do it fully.") is not acceptable.

As I asked you privately on Zulip, it is necessary that you revise your communication style so that you do not operate with accusations and similar rhetorics of blame and intimidation. Such tactics are not welcome here.

Reply:
I'm not really familiar with the functionality discussed here. The only thing I can do is moderate the conflict (if my understanding of it is good enough). Could this be a compromise: we leave the example code-block as is and add a note that *some distributions no longer ship `importlib_resources` for good reasons*. Ideally we should add some `URL` pointers that address these *good reasons*.

Reply:
> I'm not really familiar with the functionality discussed here. The only thing I can do is moderate the conflict (if my understanding of it is good enough). Could this be a compromise: we leave the example code-block as is and add a note that _some distributions no longer ship `importlib_resources` for good reasons_. Ideally we should add some `URL` pointers that address these _good reasons_.

Sorry I missed this before.

It seems we don't agree whether to recommend or not using `importlib_resources`. In this
context, I think merging this nice PR without this small snippet is a good way to go, and I thought that's what @mkoeppe was proposing. That's why I gave it positive review conditional on this.

Reply:
I'll split out the improvement of this section of the manual (orthogonal to the goal of this PR) out to a separate PR and will take @soehms's suggestion into account there.

Reply:
Done now in the rebased branch.

Reply:
... and in #37654.